src/front/spv/flow.rs

```diff
@@ -180,13 +183,28 @@ impl FlowGraph {
         }
     }
 
-    /// TODO
     /// Removes OpPhi instructions from the control flow graph and turns them into ordinary variables.
     ///
     /// Phi instructions are not supported inside Naga nor do they exist as instructions on CPUs. It is neccessary
     /// to remove them and turn into ordinary variables before converting to Naga's IR and shader code.
-    pub fn remove_phi_instructions() {
-        unimplemented!();
+    pub fn remove_phi_instructions(
+        &mut self,
+        lookup_expression: &FastHashMap<spirv::Word, LookupExpression>,
+    ) {
+        for node_index in self.flow.node_indices() {
+            for phi in self.flow[node_index].phis.clone().iter() {
```

Comment:
we should avoid a `Vec` clone here for each and every node. One way to do this would be:
```rust
let phis = mem::replace(&mut self.flow[node_index].phis, Vec::new());
for phi in phis.iter() {
   ...
}
self.flow[node_index].phis = phis;
```